huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_datasets.go

```diff
@@ -0,0 +1,255 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"type": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"output_path": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"data_source": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data_type": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+
+									"path": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"with_column_header": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"schemas": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"labels": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"property_color": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"property_shape": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"property_shortcut": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"status": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"data_format": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceDatasetsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ModelArtsV2Client(region)
+	if err != nil {
+		return diag.Errorf("Error creating ModelArts v2 client, err=%s", err)
+	}
+
+	opts := dataset.ListOpts{
+		WithLabels:    true,
+		SearchContent: d.Get("name").(string),
+	}
+
+	if v, ok := d.GetOk("type"); ok {
+		dataType := v.(int)
+		opts.DatasetType = &dataType
+	}
+
+	page, err := dataset.List(client, opts)
+	if err != nil {
+		return fmtp.DiagErrorf("Error querying ModelArts datasets: %s ", err)
+	}
+
+	p, err := page.AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf("Error querying ModelArts datasets: %s", err)
+	}
```

Comment:
The content of Error format for these return information are same, maybe you can try this:
```
pages, err := dataset.List(client, opts).AllPages()
```

Reply:
return two values, cannot in a line